packages/lib/src/interactions/InteractionsProvider.tsx

```diff
@@ -45,28 +42,12 @@ function InteractionsProvider(props: { children: ReactNode }) {
   const shouldInteract = useCallback(
     (interactionId: string, event: MouseEvent | WheelEvent) => {
       const registeredInteractions = [...interactionMap.values()];
-
-      function isButtonPressed(button: MouseButton | MouseButton[] | 'Wheel') {
-        if (event instanceof WheelEvent) {
-          return button === 'Wheel';
-        }
-
-        return Array.isArray(button)
-          ? button.includes(event.button)
-          : event.button === button;
-      }
-
-      function areKeysPressed(keys: ModifierKey[]) {
-        return keys.every((k) => event.getModifierState(k));
-      }
-
       if (!interactionMap.has(interactionId)) {
         throw new Error(`Interaction ${interactionId} is not registered`);
       }
 
       const matchingInteractions = registeredInteractions.filter(
-        ({ modifierKeys: keys, button, disabled }) =>
-          !disabled && isButtonPressed(button) && areKeysPressed(keys),
+        (interaction) => interaction.matches(event),
```

Comment:
The logic of `isButtonPressed` and `areKeysPressed` is now encapsulated in the `Interaction` class.